mail/mailer.go

```diff
@@ -6,22 +6,27 @@
 package mail
 
 import (
+	"net"
 	"net/smtp"
 )
 
-// Mailer defines a mail transfer agent to use for sending mail
-type Mailer struct {
+type Mailer interface {
+	SendMail([]string, string) error
+}
+
+// MailerImpl defines a mail transfer agent to use for sending mail
+type MailerImpl struct {
 	Server string
 	Port   string
 	Auth   smtp.Auth
 	From   string
 }
 
-// NewMailer constructs a Mailer to represent an account at a particular mail
+// NewMailer constructs a Mailer to represent an account on a particular mail
 // transfer agent.
-func NewMailer(server, port, username, password string) Mailer {
+func NewMailer(server, port, username, password string) MailerImpl {
```

Comment:
Go idiom is to just call this function New and the package mailer, so this would be called like mailer.New().